File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ANType13RecordCreateForm.vb
Imports System
Imports Neurotec.Biometrics.Standards
Imports Neurotec.IO
Imports Neurotec.Images

Namespace RecordCreateForms

	Partial Public Class ANType13RecordCreateForm
		Inherits ANRecordCreateForm
#Region "Public constructor"

		Public Sub New()
			InitializeComponent()
			For Each value As Object In System.Enum.GetValues(GetType(BdifFPImpressionType))
				cbFpImpressionType.Items.Add(value)
			Next value
			cbFpImpressionType.SelectedIndex = 0
		End Sub

#End Region

#Region "Private properties"

		Private ReadOnly Property ImpressionType() As BdifFPImpressionType
			Get
				Return CType(cbFpImpressionType.SelectedItem, BdifFPImpressionType)
			End Get
		End Property

#End Region

#Region "Protected methods"

		Protected Overrides Function OnCreateRecord(ByVal template As ANTemplate) As ANRecord
			Dim record As ANType13Record
			If imageLoader.CreateFromImage Then
				record = New ANType13Record(ANTemplate.VersionCurrent, Idc, ImpressionType, imageLoader.Src, imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, imageLoader.Image)
			ElseIf imageLoader.CreateFromData Then
				Using image As NImage = NImage.FromMemory(imageLoader.ImageData)
					record = New ANType13Record(ANTemplate.VersionCurrent, Idc, ImpressionType, imageLoader.Src, imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, image) With {.HorzLineLength = imageLoader.Hll, .VertLineLength = imageLoader.Vll, .HorzPixelScale = imageLoader.Hps, .VertPixelScale = imageLoader.Vps, .BitsPerPixel = imageLoader.Bpx}
				End Using
			Else
				Throw New NotSupportedException()
			End If

			template.Records.Add(record)
			Return record
		End Function

#End Region
	End Class
End Namespace

File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ANType13RecordCreateForm.cs
using System;
using Neurotec.Biometrics.Standards;
using Neurotec.IO;
using Neurotec.Images;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ANType13RecordCreateForm : ANRecordCreateForm
	{
		#region Public constructor

		public ANType13RecordCreateForm()
		{
			InitializeComponent();
			foreach (object value in Enum.GetValues(typeof(BdifFPImpressionType)))
			{
				cbFpImpressionType.Items.Add(value);
			}
			cbFpImpressionType.SelectedIndex = 0;
		}

		#endregion

		#region Private properties

		private BdifFPImpressionType ImpressionType
		{
			get
			{
				return (BdifFPImpressionType)cbFpImpressionType.SelectedItem;
			}
		}

		#endregion

		#region Protected methods

		protected override ANRecord OnCreateRecord(ANTemplate template)
		{
			ANType13Record record;
			if (imageLoader.CreateFromImage)
			{
				record = new ANType13Record(ANTemplate.VersionCurrent, Idc, ImpressionType, imageLoader.Src, imageLoader.ScaleUnits,
					imageLoader.CompressionAlgorithm, imageLoader.Image);
			}
			else if (imageLoader.CreateFromData)
			{
				using (NImage image = NImage.FromMemory(imageLoader.ImageData))
				{
					record = new ANType13Record(ANTemplate.VersionCurrent, Idc, ImpressionType, imageLoader.Src, imageLoader.ScaleUnits,
						imageLoader.CompressionAlgorithm, image)
					{
						HorzLineLength = imageLoader.Hll,
						VertLineLength = imageLoader.Vll,
						HorzPixelScale = imageLoader.Hps,
						VertPixelScale = imageLoader.Vps,
						BitsPerPixel = imageLoader.Bpx
					};
				}
			}
			else
			{
				throw new NotSupportedException();
			}

			template.Records.Add(record);
			return record;
		}

		#endregion
	}
}
